### PR TITLE
[FSSDK-8823] chore: prepare for release 4.0.0

### PR DESCRIPTION
**Pull request by pulak-opti**
## Summary
- prepare for release 4.0.0
- use go-sdk v2

## Issues
- [FSSDK-8823](https://jira.sso.episerver.net/browse/FSSDK-8823)

### CODE REVIEW

**Review by mikechu-optimizely — APPROVED** [review]
LGTM, but Do we need to update the paths in the Makefile?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,48 @@ All notable changes to this project will be documented in this file.
...
+
+- ODPManager in the SDK is enabled by default. Unless an ODP account is integrated into the Optimizely projects, most ODPManager functions will be ignored. If needed, ODPManager can be disabled when OptimizelyClient is instantiated. From Agent, it can be switched off from config.yaml or env variables.
+- Updated go-sdk version to v2.0.0 with module path github.com/optimizely/go-sdk/v2
+
+### Functionality Enhancement
+* Updated openapi schema to 3.1.0. ([#392](https://github.com/optimizely/agent/pull/392))
+* Added support for prometheus metrics. ([#348](https://github.com/optimizely/agent/pull/348))
+* Github Issue template is udpated. ([#396](https://github.com/optimizely/agent/pull/396))
+* Updated go version to 1.21. ([#398](https://github.com/optimizely/agent/pull/398))
+* Added OpenTelemetry Tracing Support. ([#400](https://github.com/optimizely/agent/pull/400))
```
Comment (jaeopt) [inline CHANGELOG.md:43]: This should be listed as "New Fatures"? Notification consistency and datafile syncer also?

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good! A suggestion for CHANGELOG.

**Comment by pulak-opti** [comment]
> LGTM, but Do we need to update the paths in the Makefile?

Yes, good catch, thanks

File: CHANGELOG.md (hunk 1 above)
Comment (pulak-opti) [reply CHANGELOG.md:47]: Actually Notification consistency was a bug, updated accordingly, Please check

**Review by jaeopt — APPROVED** [review]
LGTM

**Pull request merged**